sdks/python/apache_beam/io/iobase.py

```diff
@@ -1180,15 +1180,47 @@ def check_done(self):
     raise NotImplementedError
 
   def try_split(self, fraction_of_remainder):
+    """Splits current restriction based on fraction_of_remainder.
+
+    Invoked when SDK receiving ProcessBundleSplitRequest during processing
+    bundle.
+
+    Args:
+      fraction_of_remainder: a fraction of (cur_pos, stop_pos).
+
+    Returns: ``None`` when current restriction has been checkpointed, or
+    split_point is out of current restriction range. Otherwise, return
+    ((start_pos, split_pos), (split_pos, stop_pos)).
+
+    ** Thread safety **
+    Accessing to position and checkpoint status should be guarded by a single
+    lock object.
+    """
     raise NotImplementedError
 
   def try_claim(self, position):
+    """ Claims position as current_position.
```

Comment:
Try using the javadoc from https://github.com/apache/beam/blob/c0de4570b186793ff477f4a5bd86cc27191e2999/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/RestrictionTracker.java#L28 to write this version.